source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -122,13 +118,14 @@ Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, b
         fmt::format("fault.http.{}.abort.http_status", downstream_cluster_);
   }
 
-  maybeSetupResponseRateLimit();
+  maybeSetupResponseRateLimit(headers);
 
-  absl::optional<uint64_t> duration_ms = delayDuration();
-  if (duration_ms) {
+  absl::optional<std::chrono::milliseconds> duration = delayDuration(headers);
+  if (duration) {
```

Comment:
curious: do we have any style preferences around using the boolean override vs explicit has_value()?  I thought we went for explicit nullptr checks and has_value seems comparable but no strong feelings either way.

Reply:
I prefer `has_value()` and have indeed asked for it in the past. Will fix.